source/agora/test/Base.d

```diff
@@ -907,7 +907,103 @@ public class TestAPIManager
             ex.line = line;
             throw ex;
         }
+        retryFor(nodes[0].client.getBlocksFrom(0, 1).length > 0, 2.seconds);
     }
+
+    /*******************************************************************************
+
+        Add blocks up to the provided height
+
+        Params:
+            target_height = the target height after adding blocks.
+            enrolled_height = the height with the active enrollments
+
+    *******************************************************************************/
+
+    public void generateBlocks (Height target_height,
+        Height enrolled_height = Height(0),
+        string file = __FILE__, size_t line = __LINE__)
+    {
+        generateBlocks(target_height, enrolled_height, clients.take(GenesisValidators),
+            file, line);
+    }
+
+    /// Ditto
+    public void generateBlocks (Clients)(Height target_height,
+        Height enrolled_height = Height(0),
+        Clients clients = this.clients,
+        string file = __FILE__, size_t line = __LINE__)
+    {
+        assert(target_height >= enrolled_height,
+            format!"[%s:%s] Target height %s is not at least enroll height %s"
+                (file, line, target_height, enrolled_height));
+        iota(target_height - clients[0].getAllBlocks()[$ -1].header.height)
+            .each!(_ => addBlock(enrolled_height, clients, file, line));
+    }
+
+     /*******************************************************************************
+
+        Add a block
+
+        Params:
+            target_height = the target height after adding blocks.
+            enrolled_height = the height with the active enrollments
```

Comment:
Again it should state what the effect is of changing these parameters. `Add a block` is not very good documentation.

Reply:
Updated